src/transformStateWithClones.js

```
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let modifyState = { ...state };
+  const newArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        modifyState = { ...modifyState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        modifyState = { ...modifyState };
+
+        for (const key of action.keysToRemove) {
+          delete modifyState[key];
+        }
+        break;
+
+      case 'clear':
+        modifyState = {};
+        break;
+
+      default:
+        modifyState = { ...modifyState };
```

This violates the requirement: "Unknown action.type should trigger the default case (e.g., throw or handle an error)." The default case should throw an error for unknown action types instead of just cloning the state.